ortho_seq_code/plotclass.py

```diff
@@ -0,0 +1,326 @@
+                x_flat[x_flat > span[1]] = 0
+            else:
+                x[x > span] = 0
+                x_flat[x_flat > span] = 0
+        elif comp == "<>":
+            if type(span) != list:
+                print("<> and >< can only be used for list ranges.")
+                return
+            x[abs(x - np.mean(span)) >= np.mean(span)] = 0
+            x_flat[abs(x_flat - np.mean(span)) >= np.mean(span)] = 0
+        else:
+            if type(span) != list:
+                print("<> and >< can only be used for list ranges.")
+                return
+            x[abs(x - np.mean(span)) < np.mean(span)] = 0
+            x_flat[abs(x_flat - np.mean(span)) < np.mean(span)] = 0
+        print("Successfully trimmed array.")
+        self.x_flat = list(x_flat)
+        self.x = x
+
+    def histogram(
+        self,
+        bins=None,
+        site=None,
+        alphabet_item=None,
+        omit_zeroes=True,
+        bin_color=None,
+        border=True,
+        out_dir=None,
+    ):
+        if alphabet_item is not None and site is not None:
+            print("Only one of site and alphabet_item may not be null.")
+            return
+        if alphabet_item is not None and alphabet_item not in self.alphbt_input:
+            print("Alphabet item must be one of:")
+            print(self.alphbt_input)
+            return
+        if site is not None:
+            if site > self.s or site < 0:
+                print("Site must be between 0 and", str(self.s))
+                return
+        if alphabet_item is not None:
+            a = self.alphbt_input.index(alphabet_item) * self.s
+            x_red = y.x_flat[a : a + self.d]
+        elif site is not None:
+            x_red = x[site]
+        else:
+            x_red = self.x_flat
+        if omit_zeroes:
+            lst = list(x_red)
+            x_red = np.array(lst[lst != 0])
+        if border:
+            if bins is not None:
+                width = 1 - bins / 200
+            else:
+                width = 1 - len(x_red) / 1200
+        else:
+            width = 0
+        plt.hist(x_red, bins=bins, color=bin_color, lw=width, ec="black")
+        if out_dir is not None:
+            path_sav = "rFon1D_hist_" + str(ylab) + ".png"
+            path_sav = path_sav.replace(" ", "_")
+            figure.savefig(os.path.join(str(out_dir), path_sav), dpi=400)
+            print(
+                "saved regression graph as", str(os.path.join(str(out_dir), path_sav)),
+            )
+        elif self.out_dir is not None:
+            path_sav = "rFon1D_hist_" + str(ylab) + ".png"
+            path_sav = path_sav.replace(" ", "_")
+            figure.savefig(os.path.join(str(self.out_dir), path_sav), dpi=400)
+            print(
+                "saved regression graph as",
+                str(os.path.join(str(self.out_dir), path_sav)),
+            )
+
+    def set_out_dir(new_out_dir):
+        self.out_dir = new_out_dir
+
+
+def cli(command):
```

Comment:
maybe change this to `plotclass`? or some other name as we wanna stay away from using "cli" to keep things as descriptive as possible